src/components/AssetCard/utils.tsx

```diff
@@ -0,0 +1,227 @@
+    } else if (hasOnlyListings && asset.listings && asset.listings > 1) {
+      info.extraInformation = getAssetListingsRangeInfoText(asset, translations)
+    } else {
+      // has both minting and listings
+      if (hasRangeApplied) {
+        info.price = asset.minPrice ?? asset.price
+        if (appliedFilters.minPrice) {
+          const isMintingLessThanMinPriceFilter = BigNumber.from(
+            asset.price
+          ).lt(ethers.utils.parseUnits(appliedFilters.minPrice))
+          info.extraInformation = isMintingLessThanMinPriceFilter
+            ? getAlsoAvailableForMintingText(
+                asset,
+                translations.available_for_mint
+              )
+            : null
+        }
+      } else {
+        const mintIsNotCheapestOption = BigNumber.from(asset.price).gt(
+          BigNumber.from(asset.minPrice)
+        )
+        if (mintIsNotCheapestOption) {
+          info.extraInformation = getAlsoAvailableForMintingText(
+            asset,
+            translations.available_for_mint
+          )
+        }
+      }
+    }
+    return info
+  } else if (sortBy === CatalogSortBy.MOST_EXPENSIVE) {
+    const info: CatalogCardInformation = {
+      action: hasRangeApplied
+        ? translations.most_expensive_range
+        : translations.most_expensive,
+      actionIcon: null,
+      price: asset.price,
+      extraInformation: getAssetListingsRangeInfoText(asset, translations)
+    }
+
+    const isMintingGreaterThanMaxListingPrice =
+      !!asset.maxListingPrice &&
+      BigNumber.from(asset.price).gt(BigNumber.from(asset.maxListingPrice))
+
+    info.price =
+      getIsMintPriceInRange(asset, appliedFilters) &&
+      isMintingGreaterThanMaxListingPrice
+        ? asset.price
+        : asset.maxListingPrice ?? asset.price
+
+    return info
+  }
+
+  // rest of filter without label logic
+  const info: CatalogCardInformation = {
+    action: '',
+    actionIcon: null,
+    price: asset.price,
+    extraInformation: null
+  }
+
+  if (hasOnlyMint) {
+    info.action = translations.available_for_mint
+    info.actionIcon = 'mintingIcon'
+  } else if (hasOnlyListings) {
+    ;(info.action = hasRangeApplied
+      ? translations.available_listings_in_range
+      : translations.cheapest_listing),
+      (info.price =
+        asset.listings &&
+        asset.listings > 1 &&
+        asset.minListingPrice !== asset.maxListingPrice
+          ? hasRangeApplied
+            ? getListingsRangePrice(asset)
+            : asset.minListingPrice ?? ''
+          : asset.minPrice ?? '')
+  } else {
+    // both mint and listings available
+
+    if (hasRangeApplied) {
+      const isMintInRange = getIsMintPriceInRange(asset, appliedFilters)
+      info.action = isMintInRange
+        ? translations.available_for_mint
+        : translations.available_listings_in_range
+      info.price = isMintInRange ? asset.price : asset.minListingPrice ?? ''
+      info.actionIcon = isMintInRange ? 'mintingIcon' : null
```

Comment:
This can be simplified using the `??`operator, would you mind changing it?
```suggestion
      info.actionIcon = isMintInRange ?? 'mintingIcon'
```
The same goes for the line below.